swift/LottieRenderer.swift

```diff
@@ -0,0 +1,65 @@
+import CoreGraphics
+
+import thorvg
+
+/// Shorthand alias for the buffer type, representing image pixel data in a mutable pointer to UInt32.
+public typealias Buffer = UnsafeMutablePointer<UInt32>
+
+/// Object responsible for rendering a Lottie animation using ThorVG.
+public class LottieRenderer {
+    private let lottie: Lottie
+    private let engine: Engine
+    private let canvas: Canvas
+
+    /// Initializes the LottieRenderer with a specific Lottie object, engine, size, buffer, and stride.
+    /// - Parameters:
+    ///   - lottie: The `Lottie` object containing the animation to render.
+    ///   - engine: An optional `Engine` object to use. If not provided, the default engine configuration is used.
+    ///   - size: The size of the rendering canvas. This size determines the final size of the rendered Lottie content.
+    ///   - buffer: A buffer to hold the rendered pixel data.
+    ///   - stride: The number of bytes in a row of the buffer.
+    public init(
+        _ lottie: Lottie,
+        engine: Engine = .default,
+        size: CGSize,
+        buffer: Buffer,
+        stride: Int
+    ) {
+        self.lottie = lottie
+        self.engine = engine
+        self.canvas = Canvas(size: size, buffer: buffer, stride: stride)
+    }
```

Comment:
Just a heads up: I've moved the `Lottie` instance out of the `render` function and into the `LottieRenderer` initialiser. 

Previously, we actually only supported rendering a single Lottie anyways due to the fact that we only push to the Canvas once (further down - line 57), which was actually a bug of the old implementation.

Reply:
We still have a 1:1 mapping of `Lottie` instance to `LottieRenderer` instance, so I don't think this change will immediately see performance increases across multiple Lotties. But I will check 👍